edalize/edatool.py

```diff
@@ -104,6 +104,15 @@ def get_doc(cls, api_ver):
         else:
             logger.warning("Invalid API version '{}' for get_tool_options".format(api_ver))
 
+    def check_args(self, unknown):
+        # If a tool is using subtools some of the argument may be
+        # parsed by the subtool. This function is used to check if
+        # all the provided arguments were correct. A tool can override
+        # this function to provide custom args checking logic.
+
+        if unknown:
+            raise Exception(f'Unknown command line option {unknown[0]}')
```

Comment:
From #144 we plan to support Python 3.5, so I'm afraid you'll have to use `.format` instead of f-strings. Also, if this is an abstract method would just raising `NotImplementedError` if it's called be the most informative exception?

Reply:
Done